### ISSUE TITLE
find_nodes_in_area expansion

### ISSUE THREAD

**Issue by tenplus1** [minetest/minetest#2326]
I like using the find_nodes_in_area command for certain routines which returns either a list of positions or # of nodes found, but sometimes I wish it would return how many of EACH node it has found ..or.. a list of nodenames to go with the positions...

Feature 1:
```
nodes, dirt, stone = minetest.find_nodes_in_area(minp, maxp, {"default:dirt", "default:stone"})
print ("dirt = ", dirt, "stone = ", stone)
```
Feature 2:
```
nodes = minetest.find_nodes_in_area(minp, maxp, {"default:dirt", "default:stone"})
local dirt, stone = 0
for _, nod in ipairs(nodes)
   if nodes[nod].nodename == "default:dirt" then dirt = dirt + 1 end
   if nodes[nod].nodename == "default:stone" then stone = stone + 1 end
end
print ("dirt = ", dirt, "stone = ", stone)
```

**Comment by PilzAdam** [comment]
This can be easily done by calling `minetest.find_nodes_in_area` two times:
```Lua
local dirt_nodes = minetest.find_nodes_in_area(minp, maxp, {"default:dirt"})
local stone_nodes = minetest.find_nodes_in_area(minp, maxp, {"default:stone"})
```

**Comment by tenplus1** [comment]
Yes, but in doing so it checks the same area twice and can cause lag on servers...  Minetest could easily check the area once and total the nodes it has found...

**Comment by MT-Modder** [comment]
I like this idea :+1:. Reducing lag is critical.

**Comment by tenplus1** [comment]
...and many mods do use this command so it will benefit a lot of servers in the process...

**Comment by TeTpaAka** [comment]
This is fixed by e50aa4ed06f36c74a892ec68d576c52ba9dc0b2c
